src/Logger.cpp

```diff
@@ -188,20 +188,18 @@ LogUnit* Logger::getLogUnit()
         }
     } else {
         std::unique_lock<std::mutex> lck(mMtx);
-        if (!mFree.empty()) {
-            log = mFree.back();
-            mFree.resize(mFree.size() - 1);
-        } else if (mLogUnitCnt < mFree.capacity()) {
-            ++mLogUnitCnt;
-        } else {
-            while (mFree.empty() && !mStop) {
-                mCond.wait(lck);
-            }
+        while (!mStop) {
             if (!mFree.empty()) {
                 log = mFree.back();
                 mFree.resize(mFree.size() - 1);
+                break;
+            } else if (mLogUnitCnt < mFree.capacity()) {
+                ++mLogUnitCnt;
+                break;
             } else {
-                return nullptr;
```

Comment:
The only condtion arrive here is mStop==true, so we should abandon this log then quit asap

Reply:
I think that two queues should each have their own locks and condition variables instead of sharing the same lock and the same condition variable.